src/main/java/ru/yandex/clickhouse/response/ByteFragmentUtils.java

```diff
@@ -181,7 +193,29 @@ public static Object parseArray(ByteFragment value, Class elementClass) {
                 } else if (elementClass == BigInteger.class){
                     BigInteger bigIntegerValue = new BigInteger(fragment.asString(true));
                     java.lang.reflect.Array.set(array, index++, bigIntegerValue);
-                } else {
+                } else if (elementClass == Float.class) {
+                    Float floatValue = Float.parseFloat(fragment.asString());
+                    java.lang.reflect.Array.set(array, index++, floatValue);
+                } else if (elementClass == Double.class) {
+                    Double doubleValue = Double.parseDouble(fragment.asString());
+                    java.lang.reflect.Array.set(array, index++, doubleValue);
+                } else if (elementClass == Date.class) {
+                    Date dateValue;
+                    try {
+                        dateValue = new Date(DATE_FORMAT.parse(fragment.asString()).getTime());
+                    } catch (ParseException e) {
+                        throw new IllegalArgumentException(e);
+                    }
+                    java.lang.reflect.Array.set(array, index++, dateValue);
+                } else  if (elementClass == Timestamp.class) {
+                    Timestamp dateTimeValue;
+                    try {
+                        dateTimeValue = new Timestamp(DATE_TIME_FORMAT.parse(fragment.asString()).getTime());
```

Comment:
fragment here represents datetime string in server timezone, I think it is incorrect to parse it with UTC time zone.